sdk/servicebus/service-bus/src/session/messageSession.ts

```diff
@@ -761,6 +761,27 @@ export class MessageSession extends LinkEntity<Receiver> {
     }
   }
 
+  /**
+   * To be called when connection is disconnected to gracefully close ongoing receive request.
+   * @param connectionError - The connection error if any.
+   */
+  async onDetached(connectionError: AmqpError | Error | undefined): Promise<void> {
+    if (this._batchingReceiverLite.isReceivingMessages) {
+      if (connectionError == null) {
+        connectionError = new Error(
+          "Unknown error occurred on the AMQP connection while receiving messages."
+        );
+      }
+      this._batchingReceiverLite.terminate(connectionError);
+
+      // .close() also calls terminate(without error) but the _closeHandler would be undefined by then,
+      // hence no effect for that "terminate" call.
+      await this.close(); // TODO: Based on how the streaming receivers will be handled, this `.close` call can be called conditionally or moved around accordingly
```

Comment:
So if we extended `MessageSession.close()` to just take in an optional error then we wouldn't need to do this manual `_batchingReceiverLite.terminate` above? 

It seems cleaner to let close() do all the work rather than splitting it up.

Reply:
Sounds good. I can attempt that and run the tests.

Reply:
Done